src/utils/getAreasList/getAreasList.js

```diff
@@ -21,26 +20,33 @@ export type TAreasList = {
 }
 
 export default function getAreasList(props: TProps): TAreasList {
-  const areas = Object.keys(props).reduce(
+  const propKeys = Object.keys(props)
+
+  /*
+    if the propValue is not a string it is disqualified from being a template
+    from the very get go, mini-optimization there + it removes the need to
+    type-cast the value passed to sanitizeTemplateString since flow can now
+    infer that the value can only be a string otherwise it would have been
+    filtered out already
+  */
+  const templatePropKeys = propKeys.filter(
```

Comment:
Please make this function pure and remove from the `getAreasList` declaration scope.
That is to prevent re-declarations on each `getAreasList` call. Also, pure functions are generally easier to test.

Reply:
I actually first did that. I made a function called 'isTemplateProp' but it literally was just one line of code and I thought it wasn't worth it but I can definitely extract that into its own function + you're right about the redeclaration + this check is probably gonna be needed in multiple places accross this project.

Reply:
Ooooh the whole template filtering function, will do sir, that is a great idea :)

Reply:
Could you please give it a try to design the function that accepts the props object, and returns the filtered object (or an array of key names) with "template*" props only? It would then integrate nicely in the function composition I've described above.

```ts
function filterTemplateProps(props: TProps): TTemplateProps {}
```

We can then call `filterTemplateProps` functions as the first step of `getAreasList`, getting the object we would iterate over. What do you think about that?

Reply:
Yea that sounds perfect I'll do that when I find the time to do it :)